### DISCUSSION TITLE
Why Is OnNavigatedTo firing in different ways

### DISCUSSION THREAD

**Discussion by GWawrzeniecki** [PrismLibrary/Prism#2508] [WPF]
Hello

I have different ViewModels which are implementing the INavigationAware interface.
In one ViewModel OnNavigatedTo is firing just after the constructor and all the binding are working properly and in the second view model the OnNavigatedTo is firing after the bindings which results in binding errors.
It's really disappointing.

**Comment by brianlagunas** [comment]
Without any context to the code you are using, which is usually the first source of any potential issues, Prism does not control when the .NET Framework bindings happen. Also, having OnNavigated to fired after the bindings does not cause a binding error. Binding errors occur when a binding cannot be found on the data context. So either the data context has not been set yet, or the property doesn't exist on the data context.

**Reply by GWawrzeniecki** [reply]
First of all, thx for the answer!

I know that help without any context is very hard, maybe we can solve issue without giving context.
So my question is when exactly OnNavigated Error is firing? I suppose its firing some time after constructor, am I right?
So the bindings and the OnNavigated error can firing simultaneously or separately, in short randomly?

**Reply by brianlagunas** [reply]
OnNavigatedTo, as the name implies, only fires after the View/VM has been navigated to.

**Reply by GWawrzeniecki** [reply]
Okay.
My problem is as follow:
In OnNavigatedTo I load some data on which properties are using. That properties are bind to the view.
So the view is loading and I get null reference exceptions in bindings because data that I get in OnNavigatedTo isn't loaded yet. Of course, I simply get data from NavigationParameters, not from DB.

**Reply by brianlagunas** [reply]
If you are getting null reference exceptions, you need to protect your code from being null.

**Reply by GWawrzeniecki** [reply]
I could add a null check-in for all properties which are bind to the view.
So this is the only way I can solve this?

**Reply by brianlagunas** [reply]
I really have no idea what your issue is. If you are getting runtime null exceptions, then yes you have to guard against null.

**Reply by GWawrzeniecki** [reply]
View Model:
[https://github.com/GWawrzeniecki/Pawnshop/blob/master/Modules/PawnShop.Modules.Contract/ViewModels/RenewContractDataViewModel.cs](url)
View:
[https://github.com/GWawrzeniecki/Pawnshop/blob/master/Modules/PawnShop.Modules.Contract/Views/RenewContractData.xaml](url)

Here you have the code where I have this issue

**Reply by brianlagunas** [reply]
You are using object instances (and properties off that object) in your getters of other properties without checking for null.

**Reply by GWawrzeniecki** [reply]
okay, so I will just add null checks.
Thank you very much for your valuable time.
